src/core/Akka/Util/AtomicReference.cs

```diff
@@ -81,6 +81,18 @@ public bool CompareAndSet(T expected, T newValue)
             return false;
         }
 
+        /// <summary>
+        /// Returns the current value and overrides it with the new one.
+        /// </summary>
+        /// <param name="newValue">The new value</param>
+        /// <returns>The current value</returns>
+        public T GetAndSet(T newValue)
```

Comment:
```Interlocked.Exchange``` unfortunately only allows reference types  in the generic version.
Would this also work?
```
Monitor.Enter(_lock);
var current = Value;
Value = newValue;
Monitor.Exit(_lock);
return current;
```